tests/unit_test.py

```diff
@@ -65,6 +65,9 @@ def test_unit_repr():
     assert repr(sc.Unit('uK/rad')) == 'Unit(1e-06*K**1*rad**-1)'
     assert repr(sc.Unit('m^2/s^3')) == 'Unit(1*m**2*s**-3)'
     assert repr(sc.Unit('1.234*kg')) == 'Unit(1.234*kg**1)'
+    assert repr(sc.Unit('degC')) == 'Unit(1*K**1, e_flag=True)'
```

Comment:
This repr will be a problem: This is what users get to see in a notebook. I think we will have to avoid referencing these internal flags as much as possible?

Reply:
How do we distinguish degrees celsius and kelvin in this case? Or sievert and gray?

Reply:
I guess we have to format it as "deg C" and "Sv"?

Reply:
But this defeats the purpose of `repr`, IMHO. Because it is supposed to show the underlying parts and not depend on aliases or other parts of the string formatter. I don't even see how we would implement special formatting for composite units.

Reply:
As long as we can avoid users seeing this all the time I agree. If we can define, e.g., `_repr_html_` and what you get from `print` or from exception messages does not include this?

Reply:
- `_repr_html_` already exists and uses `str`
- `print` uses `str`
- exceptions: depends on the implementer

So users will only see this when they use `repr` themselves

Reply:
When using `ipython` it seems to use `repr`? Is there a different helper we have to define in addition to `_repr_html_`?

Reply:
Yes, `_repr_pretty_`: https://ipython.readthedocs.io/en/stable/config/integrating.html#custom-methods
I would do that in a separate PR as this is an independent addition.

Reply:
:+1: